test/conformance/ptp/ptp.go

```diff
@@ -184,25 +186,19 @@ var _ = Describe("[ptp]", func() {
 				ptpRunningPods = testPtpRunningPods(ptpPods)
 			})
 
-			// 25729
-			It("The interfaces support ptp can be discovered correctly", func() {
+			It("The interfaces supporting ptp can be discovered correctly", func() {
 				for podIndex := range ptpRunningPods {
-					ptpSupportedInt := getPtpMasterSlaveAttachedInterfaces(ptpRunningPods[podIndex])
+					ptpSupportedInt := getPtpInterfacePerNode(ptpRunningPods[podIndex].Spec.NodeName, fullConfig.L2Config.GetPtpIfList())
 					Expect(len(ptpSupportedInt)).To(BeNumerically(">", 0), "Fail to detect PTP Supported interfaces on slave/master pods")
 					ptpDiscoveredInterfaces := ptpDiscoveredInterfaceList(utils.NodePtpDeviceAPIPath + ptpRunningPods[podIndex].Spec.NodeName)
-					for _, intfc := range ptpSupportedInt {
-						Expect(ptpDiscoveredInterfaces).To(ContainElement(intfc))
-					}
-				}
-			})
-
-			// 25730
-			It("The virtual interfaces should be not discovered by ptp", func() {
-				for podIndex := range ptpRunningPods {
-					ptpNotSupportedInt := getNonPtpMasterSlaveAttachedInterfaces(ptpRunningPods[podIndex])
-					ptpDiscoveredInterfaces := ptpDiscoveredInterfaceList(utils.NodePtpDeviceAPIPath + ptpRunningPods[podIndex].Spec.NodeName)
-					for _, inter := range ptpNotSupportedInt {
-						Expect(ptpDiscoveredInterfaces).ToNot(ContainElement(inter), "The interfaces discovered incorrectly. PTP non supported Interfaces in list")
+					sort.Strings(ptpSupportedInt)
+					sort.Strings(ptpDiscoveredInterfaces)
+					logrus.Infof("Interfaces supporting ptp for node        %s: %v", ptpRunningPods[podIndex].Spec.NodeName, ptpSupportedInt)
+					logrus.Infof("Interfaces discovered by ptp API for node %s: %v", ptpRunningPods[podIndex].Spec.NodeName, ptpDiscoveredInterfaces)
+
+					// The discovered PTP interfaces should match exactly the list of interfaces calculated by test
+					for index := range ptpSupportedInt {
+						Expect(ptpSupportedInt[index]).To(Equal(ptpDiscoveredInterfaces[index]))
```

Comment:
Shouldn't you also check that `len(ptpDiscoveredInterfaces) == len(ptpSupportedInt)`?

Reply:
fixed